### PR TITLE
Bugfix/user class files

### LINKED ISSUE #9: USER - DATA
- [x] check import & exports
- properties >>
    - [x] id = number
    - [x] name = string
    - [x] pantry = array of objects
    - [x] favoriteRecipes = array
    - [x] recipesToCook = array
- methods >>
    - [x] a way to favorite and unfavorite a recipe
    - [x] a way to 'decide to cook' (push to array) a recipe that week. >>not sure why it says 'that week'??
    - [x] a way to filter favoriteRecipes by one or more tags
    - [x] a way to filter favoriteRecipes by name or ingredients

### LINKED ISSUE #10: USER - TEST
- [x] check import & exports
- [x] first make beforeEach() work ..currently that's stopping entire test from running
- [x] test all properties
- [x] test all methods

### PR DESCRIPTION
**Pull request by madhaus4**
# Description
Imported dataset correctly.
Fixed beforeEach functionality.
Changed functions in test file to ES6 functions.
Updated method name saveRecipe to favoriteRecipe.

&Left comments in the first 3 class methods, do you think I should return the arrays from the methods?  Thinking we might need them to return the arrays as well as store info...

## Type of change
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How Has This Been Tested?
- [ ] open index.html
- [x] console.log()
- [ ] dev tools

## Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/user.js [hunk 1]
```diff
@@ -1,29 +1,35 @@
...
-    this.recipesToCook.push(recipe);
-  }
-  filterRecipes(type) {
-    return this.favoriteRecipes.filter(recipe => recipe.type.includes(type));
-  }
-  searchForRecipe(keyword) {
-    return this.favoriteRecipes.filter(recipe => recipe.name.includes(keyword) || recipe.ingredients.includes(keyword));
-  }
+	removeFavoriteRecipe(recipe) {
+		let i = this.favoriteRecipes.indexOf(recipe);
```
Comment (Meggs625) [inline src/user.js:16]: I agree with your thought on these; from what I understand of the functionality, it makes sense to return something from these methods

File: test/user-test.js [hunk 2]
```diff
@@ -1,61 +1,57 @@
...
-    expect(user.filterRecipes('italian')).to.deep.equal([recipe]);
-  });
-
-  it('should be able to search recipes by name', function() {
-    user.saveRecipe(recipe);
-    expect(user.searchForRecipe('Chicken Parm')).to.deep.equal([recipe]);
-  });
+import users from '../src/data/users-data.js';
+
+describe('User', () => {
```
Comment (Meggs625) [inline test/user-test.js:6]: The tests look great! If we have the time, maybe we can circle back and explore more of the 'sad paths'.

**Review by Meggs625 — APPROVED** [review]

File: src/user.js (hunk 1 above)
Comment (madhaus4) [reply src/user.js:16]: ok, great!  i'll comment those in then

File: test/user-test.js (hunk 2 above)
Comment (madhaus4) [reply test/user-test.js:6]: yes! i totally agree :)

**Review by davidleach724 — APPROVED** [review]

**Pull request merged**